src/transport/SessionManager.cpp

```diff
@@ -573,20 +579,146 @@ void SessionManager::OnMessageReceived(const PeerAddress & peerAddress, System::
         }
         else
         {
-            SecureUnicastMessageDispatch(partialPacketHeader, peerAddress, std::move(msg));
+            SecureUnicastMessageDispatch(partialPacketHeader, peerAddress, std::move(msg), ctxt);
+        }
+    }
+    else
+    {
+        UnauthenticatedMessageDispatch(partialPacketHeader, peerAddress, std::move(msg), ctxt);
+    }
+}
+
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+void SessionManager::HandleConnectionReceived(Transport::ActiveConnectionState * conn)
+{
+    Transport::AppConnCallbackCtxt * appConnCbCtxt = nullptr;
+    VerifyOrReturn(conn != nullptr);
+
+    appConnCbCtxt = conn->mAppState;
+    VerifyOrReturn(appConnCbCtxt != nullptr);
+
+    if (appConnCbCtxt->connReceivedCb != nullptr)
+    {
+        ChipLogProgress(Inet, "Calling Connection Received callback");
+        appConnCbCtxt->connReceivedCb(conn);
+    }
+}
+
+void SessionManager::HandleConnectionComplete(Transport::ActiveConnectionState * conn, CHIP_ERROR conErr)
+{
+    Transport::AppConnCallbackCtxt * appConnCbCtxt = nullptr;
+    VerifyOrReturn(conn != nullptr);
+
+    appConnCbCtxt = conn->mAppState;
+    VerifyOrReturn(appConnCbCtxt != nullptr);
+
+    if (appConnCbCtxt->connCompleteCb != nullptr)
+    {
+        ChipLogProgress(Inet, "Calling Connection Complete callback");
+        appConnCbCtxt->connCompleteCb(conn, conErr);
+    }
+    else
+    {
+        // TODO: Refactor to ensure that the Disconnection is for the
+        // corresponding higher layer entity that tried to set up the connection
+        ChipLogProgress(Inet, "Connection Complete callback missing");
```

Comment:
We should remove the "callback" logging, and only log "connection established", for example. We should also log the IP address and port.

Reply:
I have removed the logging for the path where the callback is called(success path), but kept the logging with added IP address/port information since the connection is going to be disconnected in this case.

Reply:
I meant that talking about the "callback" itself is much less relevant than logging the edges of important events.